src/lib/features/ai/tools/toggle-flag.ts

```diff
@@ -0,0 +1,79 @@
+import type { RunnableToolFunctionWithParse } from 'openai/lib/RunnableFunction';
+import type { IAuditUser, IUser } from '../../../types';
+import type FeatureToggleService from '../../feature-toggle/feature-toggle-service';
+
+type toggleFlagOperationalParams = {
+    featureService: FeatureToggleService;
+    auditUser: IAuditUser;
+    user: IUser;
+};
+
+type toggleFlagParams = {
+    project: string;
+    name: string;
+    environment: string;
+    enabled: boolean;
+};
+
+type toggleFlagFunctionParams = toggleFlagOperationalParams & toggleFlagParams;
+
+const toggleFlagFunction = async ({
+    featureService,
+    auditUser,
+    user,
+    project,
+    name,
+    environment,
+    enabled,
+}: toggleFlagFunctionParams) => {
+    try {
+        const response = await featureService.updateEnabled(
+            project,
+            name,
+            environment,
+            enabled,
+            auditUser,
+            user,
+            false,
+        );
+
+        return response;
+    } catch (error) {
+        return error;
+    }
+};
+
+export const toggleFlag = ({
+    featureService,
+    auditUser,
+    user,
+}: toggleFlagOperationalParams): RunnableToolFunctionWithParse<toggleFlagParams> => ({
+    type: 'function',
+    function: {
+        function: (params: toggleFlagParams) =>
+            toggleFlagFunction({ ...params, featureService, auditUser, user }),
+        name: 'toggleFlag',
+        description:
+            'Toggle a feature flag by name, project, environment, and enabled status',
+        parse: JSON.parse,
+        parameters: {
+            type: 'object',
+            properties: {
+                project: {
+                    type: 'string',
+                    description: 'The project the flag belongs to.',
+                },
+                name: { type: 'string', description: 'The name of the flag.' },
+                environment: {
+                    type: 'string',
+                    description: 'The environment in which to toggle the flag',
+                },
+                enabled: {
+                    type: 'boolean',
+                    description: 'The desired enabled status of the flag',
```

Comment:
Maybe your intent is to tackle this in the base prompt but I can see this causing confusion with constraints forcing a flag off

Reply:
I'm not sure I follow 🤔 If a flag is off because of constraints, that doesn't matter in the perspective of our assistant, because it's only concerned about the admin side of Unleash, right?

I'm curious to know what you have in mind.